### PR TITLE
SWC-6137 - Fix SynapseTable download column issues

### PR DESCRIPTION
**Pull request by nickgros**
From [Jira](https://sagebionetworks.jira.com/browse/SWC-6137):
* Fix issue where download column would appear when SQL includes DISTINCT
* Fix issue where Add to Download Cart component wouldn't appear in Datasets

Additionally:
* Fix issue where Access column wouldn't appear for datasets

### CODE REVIEW

File: src/lib/containers/AddToDownloadListV2.tsx [hunk 1]
```diff
@@ -41,6 +41,7 @@ const AddToDownloadListV2: React.FunctionComponent<
   return (
     <>
       <a
+        data-testid="AddToDownloadListV2"
```
Comment (nickgros) [inline src/lib/containers/AddToDownloadListV2.tsx:44]: I don't like using data-testid over having accessible roles, but I think making SynapseTable truly accessible and having more complete test cases is its own unit of work

File: src/lib/containers/table/SynapseTable.tsx [hunk 2]
```diff
@@ -513,29 +515,44 @@ export default class SynapseTable extends React.Component<
...
+     *
+     * PORTALS-2010:  Enhance change made for PORTALS-1973.  File specific action will only be shown for rows that represent FileEntities.
+     */
+    const isFileViewOrDataset =
+      entity &&
+      ((isEntityView(entity) && hasFilesInView(entity)) || isDataset(entity))
+
     const isShowingAccessColumn: boolean | undefined =
-      showAccessColumn && entity && isEntityView(entity) && !containsGroupBy
+      showAccessColumn && entity && isEntityViewOrDataset && rowsHaveId
```
Comment (nickgros) [inline src/lib/containers/table/SynapseTable.tsx:543]: No reason to omit this column from datasets IMO

File: src/lib/containers/table/SynapseTable.tsx [hunk 3]
```diff
@@ -261,7 +260,10 @@ export default class SynapseTable extends React.Component<
       entityIdColumnIndicies,
     )
     // also include row entity ids if this is a view (it's possible that the ID column was not selected)
-    if (isEntityView(entity)) {
+    if (
+      (isEntityView(entity) || isDataset(entity)) &&
+      data.queryResult.queryResults.rows.every(row => !!row.rowId)
+    ) {
```
[the remark this answers is not in the document]
Comment (jay-hodgson) [reply src/lib/containers/table/SynapseTable.tsx:266]: Use your isEntityViewOrDataset() here?

File: src/lib/containers/table/SynapseTable.tsx [hunk 4]
```diff
@@ -513,29 +515,44 @@ export default class SynapseTable extends React.Component<
...
 
-    const containsGroupBy = isGroupBy(lastQueryRequest.query.sql)
+    /**
+     * If all rows have an ID, then this is a view that represents individual objects in Synapse.
+     * Presence of row IDs also indicates that the query is also necessarily not summary data, e.g. the query does
+     * not include an operation like GROUP BY, DISTINCT, etc., that would cause rows to not map 1:1 to Synapse Entities
+     */
+    const rowsHaveId = data?.queryResult.queryResults.rows.every(
+      row => !!row.rowId,
+    )
```
[the remark this answers is not in the document]
Comment (jay-hodgson) [reply src/lib/containers/table/SynapseTable.tsx:525]: TableEntity rows also have ID set.  I think the comment should be slightly updated to say that "If this is a View or Dataset and all rows have an ID, ..."

File: src/lib/containers/table/SynapseTable.tsx (hunk 3 above)
[the remark this answers is not in the document]
Comment (nickgros) [reply src/lib/containers/table/SynapseTable.tsx]: Moved from local variables to private functions to reuse

**Pull request merged**